File: Cognam/Frequency.py
def sortOnFrequency(input):
    n = int(input)
    # initialise arr
    arr = []
    for i in range(1, len(input)):
        arr.append(int(input()))
    # initialise freq
    freq = {}
    for i in range(0, n):
        if arr[i] in freq:
            freq[arr[i]] += 1
        else:
            freq[arr[i]] = 1
    # sort freq
    freq = sorted(freq.items(), key=lambda x: x[1], reverse=True)
    ans = ""
    for i in range(0, len(freq)):
        ans += str(freq[i][0]) + " "
    return ans

File: Cognam/Frequency.cpp
#include <bits/stdc++.h>
#include <iostream>
#include <vector>
#include <unordered_map>
#include <algorithm>
using namespace std;

void sortOnFrequency(string &input)
{
    int n = (int)input[0];
    vector<int> arr;
    for (int i = 1; i < input.length(); i++)
    {
        arr[i] = (int)input[i];
    }
    unordered_map<int, int> mp;
    for (int i = 0; i < n; i++)
        mp[arr[i]]++;
    sort(arr.begin(), arr.end(), [&](int a, int b)
         {
        if (mp[a] == mp[b])
            return a < b;
        return mp[a] < mp[b]; });
    string ans = "";
    for (int i = 0; i < n; i++)
        ans += (char)arr[i];
    input = ans;
}

int main()
{
    int n;
    cin >> n;
    vector<int> arr(n);
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }
    vector<int> ans = sortOnFrequency(arr, n);
    for (int i = 0; i < n; i++)
    {
        cout << ans[i] << " ";
    }
    return 0;
}